thrust/testing/zip_function.cu

```diff
@@ -176,4 +176,23 @@ struct TestNestedZipFunction2
   }
 };
 SimpleUnitTest<TestNestedZipFunction2, type_list<int, float>> TestNestedZipFunctionInstance2;
-#endif // _CCCL_STD_VER
+
+#  if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
+template <class T>
+struct ProclaimReturnType
```

Comment:
Suggestion: Could you please add a comment referring to the issue that lead to this test. Something like:
```suggestion
#  if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
// Test for regression: https://github.com/NVIDIA/cccl/issues/2344
template <class T>
struct ProclaimReturnType
```
Thx!